engine/jobmaster/dm/task_manager.go

```diff
@@ -105,6 +118,10 @@ func (tm *TaskManager) UpdateTaskStatus(taskStatus runtime.TaskStatus) {
 		zap.Uint64("config_modify_revison", taskStatus.CfgModRevision),
 	)
 	tm.tasks.Store(taskStatus.Task, taskStatus)
+	if _, ok := tm.stageGauge[taskStatus.Task]; !ok {
+		tm.stageGauge[taskStatus.Task] = tm.gaugeVec.WithLabelValues(taskStatus.Task)
+	}
+	tm.stageGauge[taskStatus.Task].Set(float64(taskStatus.Stage))
```

Comment:
Where do you clear the metrics?

Reply:
I think it is cleared after the taskManager/jobMaster close. For example, full job finished in `testSimpleAllModeTask`, it is not shown when we call `metrics`.

Reply:
The behaviour you stated is done by framework's metrics factory, and there may exist another case that we should manually clear the metrics, which is user remove an upstream by changing the job config. We can only left a TODO here

Reply:
added clear https://github.com/pingcap/tiflow/pull/7594/commits/21db7aae9a06c2703640e71304cdcbf5ff3b2267